.circleci/config.yml

```diff
@@ -70,236 +76,295 @@ parameters:
           paths:
-            - ~/.cargo
+            - ~/.cargo/bin/xtask
+            - ~/.cargo/bin/xtask.exe
 
-  xtask_check_compliance:
-    parameters:
-      os:
-        type: string
+  xtask_lint:
     steps:
       - restore_cache:
-          name: Restore .cargo
           keys:
-            - rust-cargo-<< pipeline.parameters.cache_version >>-xtask-<< parameters.os >>-{{ checksum "Cargo.lock" }}-{{ checksum "xtask/Cargo.lock" }}
-            - rust-cargo-<< pipeline.parameters.cache_version >>-xtask-<< parameters.os >>
-      - install_extra_tools:
-          os: << parameters.os >>
-      # cargo-deny fetches a rustsec advisory DB, which has to happen on github.com over https
-      - run: git config --global --unset-all url.ssh://git@github.com.insteadof
-      - run: cargo xtask check-compliance
+            - "<< pipeline.parameters.merge_version >>-lint"
+      - run: xtask lint
+      - save_cache:
+          key: "<< pipeline.parameters.merge_version >>-lint"
+          paths:
+            - target
 
-  windows_test_workspace:
+  xtask_check_compliance:
     steps:
       - restore_cache:
-          name: Restore .cargo
           keys:
-            - rust-cargo-<< pipeline.parameters.cache_version >>-test-windows-{{ checksum "Cargo.lock" }}
-            - rust-cargo-<< pipeline.parameters.cache_version >>-test-windows
-      - run: cargo test --workspace --locked
+            - "<< pipeline.parameters.merge_version >>-compliance"
+      # cargo-deny fetches a rustsec advisory DB, which has to happen on github.com over https
+      - run: git config --global --unset-all url.ssh://git@github.com.insteadof
+      - run: xtask check-compliance
       - save_cache:
-          name: Save .cargo
-          key: rust-cargo-<< pipeline.parameters.cache_version >>-test-windows-{{ checksum "Cargo.lock" }}
+          key: "<< pipeline.parameters.merge_version >>-compliance"
           paths:
-            - C:\\Users\\circleci\.cargo
+            - target
 
-  test_workspace:
+  xtask_test:
     parameters:
-      os:
-        type: string
-      cargo_test_args:
+      variant:
         type: string
-        default: "--"
-      # additional cache key to have a cache for test_updated
-      cache_key:
-        type: string
-        default: ""
+        default: "default"
     steps:
       - restore_cache:
-          name: Restore .cargo
           keys:
-            - rust-cargo-<< pipeline.parameters.cache_version >>-test-<< parameters.os >><<parameters.cache_key>>-{{ checksum "Cargo.lock" }}
-            - rust-cargo-<< pipeline.parameters.cache_version >>-test-<< parameters.os >><<parameters.cache_key>>
-
-      # As of rustc 1.61.0, must limit the number of linux jobs or we run out of memory (large executor/8GB)
-      - when:
-          condition:
-            equal: [linux_amd, << parameters.os >>]
-          steps:
-            - run: cargo test --jobs 4 --workspace --locked << parameters.cargo_test_args >> --test-threads=6
```

Comment:
It looks like we have lost control over the number of jobs by platform with this set of changes. We previously added that to stop CI jobs from crashing by starting the number of jobs to match the number of CPUs in the executor. Is this no longer a problem?

Reply:
Same as above, let's move this logic into xtask?

Reply:
We decided to pass the parameters through xtask